src/app/workflow/view/view.service.ts

```diff
@@ -130,8 +130,9 @@ export class ViewService {
    * @memberof ViewService
    */
   private requestDOISuccess(version: WorkflowVersion, workflowVersions: Array<WorkflowVersion>): void {
-    const newSelectedVersion = workflowVersions.find(v => v.id === version.id);
-    this.workflowService.setWorkflowVersion(newSelectedVersion);
+    const workflow = { ...this.workflowQuery.getActive() };
```

Comment:
Same questions/comment as on line 56.

Reply:
Yeah, almost same answer except this one can take longer to eventually fire and so could lead to problems more often.